integ-test/src/test/java/org/opensearch/sql/sql/TextTypeIT.java

```diff
@@ -0,0 +1,60 @@
+  /*
+   * Copyright OpenSearch Contributors
+   * SPDX-License-Identifier: Apache-2.0
+   */
+
+package org.opensearch.sql.sql;
+
+  import org.junit.Test;
+  import org.opensearch.sql.legacy.SQLIntegTestCase;
+  import java.io.IOException;
+
+
+  import static org.opensearch.sql.legacy.TestsConstants.TEST_INDEX_TEXTKEYWORD;
+  import static org.opensearch.sql.util.MatcherUtils.schema;
+  import static org.opensearch.sql.util.MatcherUtils.verifySchema;
+
+  public class TextTypeIT extends SQLIntegTestCase {
```

Comment:
Are there any other tests that you tried?  These only look like aggregation tests, but it would be good to know if some of the following would also work: 
* `WHERE field LIKE "keyFD??"`
* `WHERE wildcard("field", "keyFD??")`
* `SELECT field LIKE "keyFD??"` 
and maybe a couple of string-like functions: 
* `SELECT LOCATE("FD", field)`
* `SELECT SUBSTRING(field, 3, 2)`

Reply:
Also try 
* `SELECT POSITION(substring IN field)` since it seems to fail for text fields (as Margarit demonstrated)

Reply:
## Completed

- [x] WHERE field LIKE "keyFD??"
- [x] WHERE wildcard("field", "keyFD??")
- [x] SELECT field LIKE "keyFD??"
- [x] SELECT LOCATE("FD", field)
- [x] SELECT SUBSTRING(field, 3, 2)
- [x] SELECT POSITION(substring IN field)

Reply:
## Results

### "select sum(int0) from %s GROUP BY data"
**Passed:**

- aggregateOnKeyword
- aggregateOnTextFieldData
- aggregateOnKeywordFieldData
- aggregateOnTextAndFieldDataNoFields


**Failed:**

- aggregateOnText
<details>
```

REPRODUCE WITH: ./gradlew ':integ-test:integTest' --tests "org.opensearch.sql.sql.TextTypeIT.aggregateOnText" -Dtests.seed=55EBCE500405B96B -Dtests.security.manager=false -Dtests.locale=lt-LT -Dtests.timezone=AET -Druntime.java=13

org.opensearch.client.ResponseException: method [POST], host [http://127.0.0.1:57066], URI [/_plugins/_sql?format=jdbc], status line [HTTP/1.1 503 Service Unavailable]
{
  "error": {
    "reason": "Error occurred in OpenSearch engine: all shards failed",
    "details": "Shard[0]: java.lang.IllegalArgumentException: Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [typeText] in order to load field data by uninverting the inverted index. Note that this can use significant memory.\n\nFor more details, please send request for Json format to see the raw response from OpenSearch engine.",
    "type": "SearchPhaseExecutionException"
  },
  "status": 400
}
java.lang.RuntimeException: org.opensearch.client.ResponseException: method [POST], host [http://127.0.0.1:57066], URI [/_plugins/_sql?format=jdbc], status line [HTTP/1.1 503 Service Unavailable]
{
  "error": {
    "reason": "Error occurred in OpenSearch engine: all shards failed",
    "details": "Shard[0]: java.lang.IllegalArgumentException: Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [typeText] in order to load field data by uninverting the inverted index. Note that this can use significant memory.\n\nFor more details, please send request for Json format to see the raw response from OpenSearch engine.",
    "type": "SearchPhaseExecutionException"
  },
  "status": 400
}
	at __randomizedtesting.SeedInfo.seed([55EBCE500405B96B:62BFA16EB15A9522]:0)
	at org.opensearch.sql.legacy.SQLIntegTestCase.executeQuery(SQLIntegTestCase.java:239)
	at org.opensearch.sql.legacy.SQLIntegTestCase.executeJdbcRequest(SQLIntegTestCase.java:244)
	at org.opensearch.sql.sql.TextTypeIT.aggregateOnText(TextTypeIT.java:36)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
	at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
	at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
	at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
	at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.base/java.lang.Thread.run(Thread.java:830)
Caused by: org.opensearch.client.ResponseException: method [POST], host [http://127.0.0.1:57066], URI [/_plugins/_sql?format=jdbc], status line [HTTP/1.1 503 Service Unavailable]
{
  "error": {
    "reason": "Error occurred in OpenSearch engine: all shards failed",
    "details": "Shard[0]: java.lang.IllegalArgumentException: Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [typeText] in order to load field data by uninverting the inverted index. Note that this can use significant memory.\n\nFor more details, please send request for Json format to see the raw response from OpenSearch engine.",
    "type": "SearchPhaseExecutionException"
  },
  "status": 400
}
	at app//org.opensearch.client.RestClient.convertResponse(RestClient.java:375)
	at app//org.opensearch.client.RestClient.performRequest(RestClient.java:345)
	at app//org.opensearch.client.RestClient.performRequest(RestClient.java:351)
	at app//org.opensearch.client.RestClient.performRequest(RestClient.java:320)
	at app//org.opensearch.sql.legacy.SQLIntegTestCase.executeQuery(SQLIntegTestCase.java:233)
	... 42 more
	Suppressed: org.opensearch.client.ResponseException: method [POST], host [http://[::1]:57065], URI [/_plugins/_sql?format=jdbc], status line [HTTP/1.1 503 Service Unavailable]
{
  "error": {
    "reason": "Error occurred in OpenSearch engine: all shards failed",
    "details": "Shard[0]: java.lang.IllegalArgumentException: Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [typeText] in order to load field data by uninverting the inverted index. Note that this can use significant memory.\n\nFor more details, please send request for Json format to see the raw response from OpenSearch engine.",
    "type": "SearchPhaseExecutionException"
  },
  "status": 400
}
		at app//org.opensearch.client.RestClient.convertResponse(RestClient.java:375)
		at app//org.opensearch.client.RestClient.performRequest(RestClient.java:345)
		... 44 more```

</details>

Reply:
### SELECT LOCATE("FD", field)
**Passed:**

- selectLocateKeyword
- selectLocateText
- selectLocateTextKeywordFieldNoFieldData
- selectLocateTypeTextFieldData
- selectLocateTextDataFieldNoFields


**Failed:**

Reply:
### SELECT POSITION(substring IN field)
**Passed:**

- selectPositionKeyword
- selectPositionText
- selectPositionTextKeywordFieldNoFieldData
- selectPositionTypeTextFieldData
- selectLocateTextDataFieldNoFields


**Failed:**

- selectPositionTextDataFieldNoFields
<details>
```
REPRODUCE WITH: ./gradlew ':integ-test:integTest' --tests "org.opensearch.sql.sql.TextTypeIT.selectPositionTextDataFieldNoFields" -Dtests.seed=55EBCE500405B96B -Dtests.security.manager=false -Dtests.locale=lt-LT -Dtests.timezone=AET -Druntime.java=13

org.opensearch.client.ResponseException: method [POST], host [http://127.0.0.1:57066], URI [/_plugins/_sql?format=jdbc], status line [HTTP/1.1 400 Bad Request]
{
  "error": {
    "reason": "Invalid SQL query",
    "details": "syntax error, expect IN, actual RPAREN textDataFieldNoFields",
    "type": "ParserException"
  },
  "status": 400
}
java.lang.RuntimeException: org.opensearch.client.ResponseException: method [POST], host [http://127.0.0.1:57066], URI [/_plugins/_sql?format=jdbc], status line [HTTP/1.1 400 Bad Request]
{
  "error": {
    "reason": "Invalid SQL query",
    "details": "syntax error, expect IN, actual RPAREN textDataFieldNoFields",
    "type": "ParserException"
  },
  "status": 400
}
	at __randomizedtesting.SeedInfo.seed([55EBCE500405B96B:CF59CDDDCCCB3873]:0)
	at org.opensearch.sql.legacy.SQLIntegTestCase.executeQuery(SQLIntegTestCase.java:239)
	at org.opensearch.sql.legacy.SQLIntegTestCase.executeJdbcRequest(SQLIntegTestCase.java:244)
	at org.opensearch.sql.sql.TextTypeIT.selectPositionTextDataFieldNoFields(TextTypeIT.java:174)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:44)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
	at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
	at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
	at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
	at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
	at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
	at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
	at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
	at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
	at org.junit.rules.RunRules.evaluate(RunRules.java:20)
	at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
	at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
	at java.base/java.lang.Thread.run(Thread.java:830)
Caused by: org.opensearch.client.ResponseException: method [POST], host [http://127.0.0.1:57066], URI [/_plugins/_sql?format=jdbc], status line [HTTP/1.1 400 Bad Request]
{
  "error": {
    "reason": "Invalid SQL query",
    "details": "syntax error, expect IN, actual RPAREN textDataFieldNoFields",
    "type": "ParserException"
  },
  "status": 400
}
	at app//org.opensearch.client.RestClient.convertResponse(RestClient.java:375)
	at app//org.opensearch.client.RestClient.performRequest(RestClient.java:345)
	at app//org.opensearch.client.RestClient.performRequest(RestClient.java:320)
	at app//org.opensearch.sql.legacy.SQLIntegTestCase.executeQuery(SQLIntegTestCase.java:233)
	... 42 more
```

</details>

Reply:
### SELECT SUBSTRING(field, 3, 2)
**Passed:**

- selectSubstringKeyword
- selectSubstringText
- selectSubstringTextKeywordFieldNoFieldData
- selectSubstringTypeTextFieldData
- selectSubstringTextDataFieldNoFields


**Failed:**

Reply:
### WHERE field LIKE "keyFD??"
**Passed:**

- whereLikeKeyword
- whereLikeText
- whereLikeKeywordFieldNoFieldData
- whereLikeTextFieldData
- whereLiketextDataFieldNoFields


**Failed:**

Reply:
### WHERE wildcard("field", "keyFD??")
**Passed:**

- whereWildcardKeyword
- whereWildcardText
- whereWildcardKeywordFieldNoFieldData
- whereWildcardTextFieldData
- whereWildcardtextDataFieldNoFields


**Failed:**

Reply:
Can you check selectPositionTextDataFieldNoFields - its giving you a parser error.

Reply:
Otherwise, this looks exclusively like an issue with aggregation on text fields.  
Strangely, I thought POSITION was going to fail on text fields, since it was failing for Margarit earlier.